### DISCUSSION TITLE
Newe run_helper.bat

### DISCUSSION THREAD

**Discussion by MillerKillerWIller** [SunOner/yolov8_aimbot#236] [General]
I must Modify your File that my System found and install the things. Maybe this helps others so.

Version 1
`
@echo off
for /f "delims=" %%a in ('py --version 2^>nul') do set "python_version=%%a"
for /f "delims=" %%a in ('py -m pip --version 2^>nul') do set "pip_version=%%a"

echo ===================================
echo Python Version on this Computer   =
echo ===================================
echo %python_version%
echo.
echo ===================================
echo Pip Version on this Computer   =
echo ===================================
echo %pip_version%
echo.
timeout /t 1 > nul
echo.
if %errorlevel% neq 0 (
echo ==========================================
echo pip is not installed. Installing...
echo ==========================================
echo.
    py -m ensurepip --default-pip
    if %errorlevel% neq 0 (
echo ============================================================================
echo Failed to install pip automatically. Please install pip manually.
echo ============================================================================
timeout /t 2 > nul
    )
)
echo ==================================
echo pip is installed. OK
echo ==================================
py -c "import streamlit" 2>nul
if %errorlevel% neq 0 (
echo ============================================================================
    echo Streamlit is not installed. Installing...
echo ============================================================================
    py -m pip install streamlit
    if %errorlevel% neq 0 (
echo ============================================================================
echo Failed to install streamlit. Please check your internet connection and try again.
echo ============================================================================
timeout /t 2 > nul
    )
)
echo ==================================
echo Streamlit is installed. OK
echo ==================================
echo.
echo.
timeout /t 2 > nul
cls
:Auswahlmenu
echo ===================================
echo [1] Install Requirements Now      =
echo [2] Install Requirements Later    =                                                                                                       
echo ===================================
echo.
set asw=0
set /p asw="Bitte eine Auswahl treffen: "
echo.
if %asw%==1 goto Rnow
if %asw%==2 goto RLater
echo.
:Rnow
echo ==================================
echo Installing Requirements.
echo ==================================
py -m pip install -r requirements.txt
echo.
echo ==================================
echo Done.
echo ==================================
timeout /t 2 > nul
cls
echo.
goto RLater

:RLater
echo ==================================
echo Starting "helper.py" over Streamlit
echo ==================================
echo.
timeout /t 1 > nul
py -m streamlit run helper.py
echo.
echo ==============
echo Done!
echo ==============
echo.
timeout /t 5 > nul
exit
`

Version 2

@echo off
for /f "delims=" %%a in ('py --version 2^>nul') do set "python_version=%%a"
for /f "delims=" %%a in ('py -m pip --version 2^>nul') do set "pip_version=%%a"

echo ===================================
echo Python Version on this Computer   =
echo ===================================
echo %python_version%
echo.
echo ===================================
echo Pip Version on this Computer   =
echo ===================================
echo %pip_version%
echo.
timeout /t 1 > nul
echo.
if %errorlevel% neq 0 (
echo ==========================================
echo pip is not installed. Installing...
echo ==========================================
echo.
    py -m ensurepip --default-pip
    if %errorlevel% neq 0 (
echo ============================================================================
echo Failed to install pip automatically. Please install pip manually.
echo ============================================================================
timeout /t 2 > nul
    )
)
echo ==================================
echo pip is installed. OK
echo ==================================
py -c "import streamlit" 2>nul
if %errorlevel% neq 0 (
echo ============================================================================
    echo Streamlit is not installed. Installing...
echo ============================================================================
    py -m pip install streamlit
    if %errorlevel% neq 0 (
echo ============================================================================
echo Failed to install streamlit. Please check your internet connection and try again.
echo ============================================================================
timeout /t 2 > nul
    )
)
echo ==================================
echo Streamlit is installed. OK
echo ==================================
echo.
echo.
timeout /t 2 > nul
cls
:Auswahlmenu
echo ===================================
echo [1] Install Requirements Now      =
echo [2] Install Requirements Later    =                                                                                                       
echo ===================================
echo.
set asw=0
set /p asw="Choice a Number: "
echo.
if %asw%==1 goto Rnow
if %asw%==2 goto RLater
echo.
:Rnow
echo ==================================
echo Installing AIMBot Requirements.
echo ==================================
py -m pip install -r requirements.txt
echo.
echo ==================================
echo Installing Main Requirements.
echo ==================================
echo Cuda 12.4 is being downloaded, and installation will begin after downloading.
echo.
powershell -command "& { (New-Object Net.WebClient).DownloadFile('https://developer.download.nvidia.com/compute/cuda/12.4.0/local_installers/cuda_12.4.0_551.61_windows.exe', 'cuda_12.4.0_551.61_windows.exe') }"
echo Run Installation
timeout /t 1 > nul
echo.
start /wait cuda_12.4.0_551.61_windows.exe
timeout /t 10 > nul
:check_installation
tasklist /FI "IMAGENAME eq cuda_12.4.0_551.61_windows.exe" 2>NUL | find /I /N "cuda_12.4.0_551.61_windows.exe" >NUL
if "%ERRORLEVEL%"=="0" (
    echo CUDA Installation is complete.
echo ==================================
echo Done.
echo ==================================
timeout /t 1 > nul
cls
) else (
    timeout /t 5 > nul
    goto check_installation
)
echo ==================================
echo Installing Torch.
echo ==================================
py -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu124
echo.
echo ==================================
echo Done.
echo ==================================
echo.
timeout /t 2 > nul
cls
echo ==================================
echo Installing Torch.
echo ==================================
py -m pip install tensorrt
echo.
echo ==================================
echo Done.
echo ==================================
echo.
timeout /t 2 > nul
cls
echo.
goto RLater

:RLater
echo ==================================
echo Starting "helper.py" over Streamlit
echo ==================================
echo.
timeout /t 1 > nul
py -m streamlit run helper.py
echo.
echo ==============
echo Done!
echo ==============
echo.
timeout /t 5 > nul
exit

**Comment by MillerKillerWIller** [comment]
[run_helper.txt](https://github.com/user-attachments/files/16258640/run_helper.txt)

**Reply by NANOBYTECOMPUTERS** [reply]
copyleft feel free to take and reuse however needed. I forgot to add a license file
